### PR TITLE
[exporter/clickhouse] Fix StartTime is not inserted for Gauge and Counter

### LINKED ISSUE #18220: ClickHouse - StartTime is not inserted for Gauge and Counter
### Component(s)

exporter/clickhouse

## Description

Hi, I tried metrics with the clickhouse collector. I saw that the column `StartTimeUnix` displayed the UNIX start date `1970`. After digging into the source, I found out that the start time is not inserted.

- (Counter) https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/main/exporter/clickhouseexporter/internal/sum_metrics.go#L129
- (Gauge) https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/main/exporter/clickhouseexporter/internal/gauge_metrics.go#L125

### Collector version

0.70.0

## Environment
OS: (e.g., "Ubuntu 20.04")
Compiler(if manually compiled): (e.g., "go 14.2")

### OpenTelemetry Collector configuration

```yaml
extensions:
receivers:
  otlp:
    protocols:
      grpc:
      http:

processors:
  batch:
  memory_limiter:
    # 75% of maximum memory up to 4G
    limit_mib: 1536
    # 25% of limit up to 2G
    spike_limit_mib: 512
    check_interval: 5s

exporters:
  clickhouse:
    endpoint: "url"
    timeout: 10s
    database: otel
    traces_table_name: otel_traces
    logs_table_name: otel_logs
    metrics_table_name: otel_metrics
    ttl_days: 7
    sending_queue:
      queue_size: 5000
    retry_on_failure:
      enabled: true
      initial_interval: 10s
      max_interval: 30s
      max_elapsed_time: 300s
#  logging:
#    loglevel: debug

service:
  pipelines:
    metrics:
      receivers: [otlp]
      processors: [memory_limiter, batch]
      exporters: [clickhouse]
```

### Log output

_No response_

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by Frapschen**
**Description:** Fix StartTime is not inserted for Gauge and Counter

fix https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/18220 co-author with @StarpTech

### CODE REVIEW

File: exporter/clickhouseexporter/internal/gauge_metrics.go [hunk 1]
```diff
@@ -122,16 +123,17 @@ func (g *gaugeMetrics) insert(ctx context.Context, db *sql.DB) error {
 			valueArgs[index+8] = model.metricDescription
 			valueArgs[index+9] = model.metricUnit
 			valueArgs[index+10] = attributesToMap(dp.Attributes())
-			valueArgs[index+11] = dp.Timestamp().AsTime().UnixNano()
-			valueArgs[index+12] = getValue(dp.IntValue(), dp.DoubleValue(), dp.ValueType())
-			valueArgs[index+13] = uint32(dp.Flags())
+			valueArgs[index+11] = dp.StartTimestamp().AsTime().UnixNano()
```
Comment (StarpTech) [inline exporter/clickhouseexporter/internal/gauge_metrics.go:126]: Note: In Gauge, I experienced that the date was incorrect. According to the spec https://opentelemetry.io/docs/reference/specification/metrics/data-model/#gauge the field is optional. Probably, it is not supported.

Comment (dmitryax) [reply exporter/clickhouseexporter/internal/gauge_metrics.go:126]: It's indeed still unclear what the start time for gauges is supposed to represent, and usually it's empty.

File: exporter/clickhouseexporter/internal/gauge_metrics.go [hunk 2]
```diff
@@ -70,6 +70,7 @@ SETTINGS index_granularity=8192, ttl_only_drop_parts = 1;
     MetricDescription,
     MetricUnit,
     Attributes,
+    StartTimeUnix,
```
Comment (dmitryax) [inline exporter/clickhouseexporter/internal/gauge_metrics.go:73]: How is the upgrade experience going to be with such a change? The table will not be recreated automatically, and the new version of the exporter won't work until the table is recreated, right?

Comment (StarpTech) [reply exporter/clickhouseexporter/internal/gauge_metrics.go:73]: Correct. What's the guarantee here in general? Do we follow semver? We could mention that breaking change in the readme.

File: exporter/clickhouseexporter/internal/gauge_metrics.go (hunk 1 above)
Comment (Frapschen) [reply exporter/clickhouseexporter/internal/gauge_metrics.go:126]: I found that all of metric type's `start_time_unix_nano`  is optinal, see https://github.com/open-telemetry/opentelemetry-proto/blob/main/opentelemetry/proto/metrics/v1/metrics.proto

File: exporter/clickhouseexporter/internal/gauge_metrics.go (hunk 2 above)
Comment (Frapschen) [reply exporter/clickhouseexporter/internal/gauge_metrics.go:73]: The table create SQL already has column `StartTimeUnix`, @dmitryax your worries won't happen.

https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/18288/files#diff-f7bdb65e4ce3de30d08997f907af2cc3ce6faf3ae319d6473d72d003936e4617L43 

https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/18288/files#diff-e10408b52409e5277dc13b218531e360d5988267347bb33b58fb28ec32b12b55L43

Comment (StarpTech) [reply exporter/clickhouseexporter/internal/gauge_metrics.go:73]: You're correct.

Comment (dmitryax) [reply exporter/clickhouseexporter/internal/gauge_metrics.go:73]: > The table create SQL already has column StartTimeUnix, @dmitryax your worries won't happen.

You pointed to this PR updating the create table query. What do you mean by "your worries won't happen"?

Comment (dmitryax) [reply exporter/clickhouseexporter/internal/gauge_metrics.go:73]: > What's the guarantee here in general? 

I believe it's ok given that the component is in alpha stability state. 

> We could mention that breaking change in the release notes.

I believe we should mark it as a breaking change in the changelog entry and say that the upgrade requires removing up upgrading the old table

Comment (StarpTech) [reply exporter/clickhouseexporter/internal/gauge_metrics.go:73]: Sounds good.

Comment (Frapschen) [reply exporter/clickhouseexporter/internal/gauge_metrics.go:73]: > You pointed to this PR updating the create table query. 

This PR not update the create table query but only change the insert table query. 

> What do you mean by "your worries won't happen"?

It means the new version of the exporter will work fine without the table is recreated.

Comment (dmitryax) [reply exporter/clickhouseexporter/internal/gauge_metrics.go:73]: Oh, Ok, I missed that. Sorry

**Review by dmitryax — APPROVED** [review]

**Pull request merged**